docs/guides/es2k/deploying-p4-programs.md

```diff
@@ -0,0 +1,57 @@
+# Deploying P4 Programs for ES2K
+
+## 1. Overview
+
+This document explains how to deploy the P4 artifacts on hardware flexible
+pipeline (FXP).
+
+Note: This document assumes that you have followed
+[Compiling P4 programs guide](https://github.com/ipdk-io/networking-recipe/blob/main/docs/guides/es2k/compiling-p4-programs.md)
+to compile and generate artifacts required for deployment.
+
+## 2. Deploying on Hardware Flexible Pipeline (FXP)
+
+Data Path Control Plane (DPCP) starts with a default P4 package. To load a
+custom P4 package follow below steps:
+
+### 2.1 Interrupt default startup routine
+
+Reboot IMC and type ``N`` when the following message is shown on IMC console::
+
+```text
+start ipumgmtd and auxiliary script [Y/N] \
+(default start both automatically after 10 seconds)?
+```
+
+### 2.2 Copy the custom P4 package
+
+Copy the custom P4 package (.pkg) in `/etc/dpcp/package` directory and
+overwrite the `default_pkg.pkg`.
+
+For example, replace `default_pkg.pkg` with `simple_l3_l4_pna.pkg`
+
+```bash
+root@mev-imc:/etc/dpcp/package# ls -lrt /etc/dpcp/package/
+total 2364
+-rw-r--r-- 1 root root  963032 Jan  1 04:56 simple_l3_l4_pna.pkg
+-rw-r--r-- 1 root root 1450456 Jun  8  2023 e2100-default-1.0.3.0.pkg
+drwxr-xr-x 2 root root       0 Jun  8  2023 runtime_files
+drwxr-xr-x 3 root root       0 Jun  8  2023 ref_pkg
+lrwxrwxrwx 1 root root      25 Jun  8  2023 default_pkg.pkg -> e2100-default-1.0.3.0.pkg
+root@mev-imc:/etc/dpcp/package# cp simple_l3_l4_pna.pkg default_pkg.pkg
+```
+
```

Comment:
After "Copy the custom P4 package" step we need to edit the 
/etc/dpcpcfg/cp_init.cfg file for following configuration
cpf_host = 4; /* 0 ~ 4. Only one CPF (Xeon 0~3, or ACC 4)
pf_allowed_to_create_p2p = 12 [12 - ACC , 8 - Host]
sem_num_pages = 25;
else DPCP service will not start

Reply:
I believe the defaults have been updated (to be confirmed)

Reply:
In latest CI release 5408, we only need to ensure cpf_host is set to 4 for ACC. (this is covered in line 52 and 53) and comments for p2p has changed and not different for ACC/Host configuration:
**pf_allowed_to_create_p2p = 12; /* indicate which PF allowed to perform p2p configuration */**

sem_num_pages should be set to 25 only for linux networking use case. Sandeep said he will cover it as part of Linux networking documentation since it's specific to that P4.